client/dashboard/domains/domain-forwarding/add.tsx

```diff
@@ -0,0 +1,50 @@
+import { useMutation } from '@tanstack/react-query';
+import { useRouter } from '@tanstack/react-router';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { domainForwardingSaveMutation } from '../../app/queries/domain-forwarding';
+import { domainRoute, domainForwardingsRoute } from '../../app/router';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import DomainForwardingForm from './form';
+import { formDataToSubmitData } from './utils';
+import type { FormData } from './form';
+import type { DomainForwardingSaveData } from '../../data/domain-forwarding';
+
+export default function AddDomainForwarding() {
+	const router = useRouter();
+	const { domainName } = domainRoute.useParams();
+	const saveMutation = useMutation( domainForwardingSaveMutation( domainName ) );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	const handleSubmit = ( formData: FormData ) => {
```

Comment:
Looks like we can move this function to the `DomainForwardingForm` itself and then we just need to pass something like `onSuccess/onError` or `successMessage/errorMessage` to the component. What do you think?

Reply:
We could also handle this differently by passing onSuccess and onError as props. I’ll leave the call up to you 🙂